### PR TITLE
Added In-place heap sort and Max. Priority Queue using java

### PR DESCRIPTION
**Pull request by Tanmay2510**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Review by atarax665 — COMMENTED** [review]
Remove all the irrelevant files.

**Comment by Tanmay2510** [comment]
Hey, i have removed the irrelevant files. Now can you review my pull request @atarax665

File: Java/ds/MaxPriorityQueue.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+//Implementing the class for Max Priority Queue which includes the required functions......
```
Comment (atarax665) [inline Java/ds/MaxPriorityQueue.java:1]: Add an introduction of what is a Max Priority Queue.

File: Java/ds/MaxPriorityQueue.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                heap.set(parentIndex,r);
+                heap.set(rightchildIndex,p);
+                parentIndex =rightchildIndex;
+                leftchildIndex = (2*parentIndex)+1;
+                rightchildIndex = (2*parentIndex)+2;
+            }
+        }
+        return remove;
+	}
+}
```
Comment (atarax665) [inline Java/ds/MaxPriorityQueue.java:100]: Add driver code and add sample input/output.

File: Java/sort/InplaceHeapSort.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+//Given an integer array of size N, sorting it using Heap Sort in descending order......
+
+public class InplaceHeapSort {
+	
+	
+	public static void inplaceHeapSort(int arr[]) {
+		
+         int n = arr.length;
```
Comment (atarax665) [inline Java/sort/InplaceHeapSort.java]: Inconsistent indentation.

File: Java/sort/InplaceHeapSort.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+//Given an integer array of size N, sorting it using Heap Sort in descending order......
+
+public class InplaceHeapSort {
+
```
Comment (atarax665) [inline Java/sort/InplaceHeapSort.java]: Remove all the unnecessary blank lines.

File: Java/sort/InplaceHeapSort.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            arr[largest] = swap; 
+  
+            // Recursively heapify the affected sub-tree 
+            heapify(arr, n, largest); 
+        } 
+		
+	}
+         
+}
+
```
Comment (atarax665) [inline Java/sort/InplaceHeapSort.java]: Add driver code and sample input/output.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Tanmay2510** [comment]
Can you review the file @atarax665? I have made the required changes in file.

**Comment by Tanmay2510** [comment]
Hey, I have updated the readme file  and made all the needy changes. Can you review my pull request @HarshCasper @atarax665 ?

**Review by atarax665 — CHANGES_REQUESTED** [review]
Please do not remove  ```.gitignore```. Add driver code.

File: Java/sort/InplaceHeapSort.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            arr[largest] = swap; 
+            heapify(arr, n, largest); 
+              } 
+	}
+  }
+// Sample Input
+// 6 
+// 2 6 8 5 4 3
+// Sample Output
+// 8 6 5 4 3 2
```
Comment (HarshCasper) [inline Java/sort/InplaceHeapSort.java]: Add them in a multi-line comment

File: Java/sort/InplaceHeapSort.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            largest = r; 
+	}
+        if (largest != i) 
+        { 
+            int swap = arr[i]; 
+            arr[i] = arr[largest]; 
+            arr[largest] = swap; 
+            heapify(arr, n, largest); 
+              } 
+	}
```
Comment (HarshCasper) [inline Java/sort/InplaceHeapSort.java]: Properly indent the Code

File: Java/ds/MaxPriorityQueue.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+// Sample Input
+// 6 4 2 12 10 
+  
+// Sample Output
+// 12 10 8 6 4 2
```
Comment (HarshCasper) [inline Java/ds/MaxPriorityQueue.java]: Add in a multi-line comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Tanmay2510** [comment]
Can you see the changes now @HarshCasper @atarax665 ?
Screenshot 2021-02-25 144518

**Comment by HarshCasper** [comment]
We are not familiar with this UI tool which you are using to visualize the Git branching and commits. Have a look [here](https://github.com/TesseractCoding/NeoAlgo/pull/2150/files) and see all the changes being pushed by you in your commits.

**Pull request closed without merging**